lib/api/controllers/documentController.js

```diff
@@ -173,6 +176,49 @@ class DocumentController extends NativeController {
     };
   }
 
+  async export (request) {
+    const { index, collection } = request.getIndexAndCollection();
+    const { size, scrollTTL, searchBody } = request.getSearchParams();
+    const format = request.getString('format', 'jsonl');
+    const fields = request.getBodyArray('fields', []);
+    const lang = request.getLangParam();
+    const separator = request.getString('separator', ',');
+
+    if (request.context.connection.protocol !== 'http') {
+      throw kerror.get('api', 'assert', 'unsupported_protocol', request.context.connection.protocol, 'document:export');
+    }
+
+    if (lang === 'koncorde') {
+      searchBody.query = await this.translateKoncorde(searchBody.query || {});
+    }
+
+    const stream = new BufferedPassThrough({ highWaterMark: 16384 });
+
+    dumpCollectionData(
```

Comment:
I think it would make more sense if the `dumpCollectionData` would returns the stream. Also it could directly returns an HttpStream